Sources/RemoteMessaging/Matchers/UserAttributeMatcher.swift

```diff
@@ -95,6 +93,73 @@ public struct MobileUserAttributeMatcher: AttributeMatching {
 
 }
 
+public struct DesktopUserAttributeMatcher: AttributeMatching {
+    private let pinnedTabsCount: Int
+    private let hasCustomHomePage: Bool
+
+    private let commonUserAttributeMatcher: CommonUserAttributeMatcher
+
+    public init(statisticsStore: StatisticsStore,
+                variantManager: VariantManager,
+                emailManager: EmailManager = EmailManager(),
+                bookmarksCount: Int,
+                favoritesCount: Int,
+                appTheme: String,
+                daysSinceNetPEnabled: Int,
+                isPrivacyProEligibleUser: Bool,
+                isPrivacyProSubscriber: Bool,
+                privacyProDaysSinceSubscribed: Int,
+                privacyProDaysUntilExpiry: Int,
+                privacyProPurchasePlatform: String?,
+                isPrivacyProSubscriptionActive: Bool,
+                isPrivacyProSubscriptionExpiring: Bool,
+                isPrivacyProSubscriptionExpired: Bool,
+                dismissedMessageIds: [String],
+                pinnedTabsCount: Int,
+                hasCustomHomePage: Bool
+    ) {
+        self.pinnedTabsCount = pinnedTabsCount
+        self.hasCustomHomePage = hasCustomHomePage
+
+        commonUserAttributeMatcher = .init(
+            statisticsStore: statisticsStore,
+            variantManager: variantManager,
+            emailManager: emailManager,
+            bookmarksCount: bookmarksCount,
+            favoritesCount: favoritesCount,
+            appTheme: appTheme,
+            daysSinceNetPEnabled: daysSinceNetPEnabled,
+            isPrivacyProEligibleUser: isPrivacyProEligibleUser,
+            isPrivacyProSubscriber: isPrivacyProSubscriber,
+            privacyProDaysSinceSubscribed: privacyProDaysSinceSubscribed,
+            privacyProDaysUntilExpiry: privacyProDaysUntilExpiry,
+            privacyProPurchasePlatform: privacyProPurchasePlatform,
+            isPrivacyProSubscriptionActive: isPrivacyProSubscriptionActive,
+            isPrivacyProSubscriptionExpiring: isPrivacyProSubscriptionExpiring,
+            isPrivacyProSubscriptionExpired: isPrivacyProSubscriptionExpired,
+            dismissedMessageIds: dismissedMessageIds
+        )
+    }
+
+    public func evaluate(matchingAttribute: MatchingAttribute) -> EvaluationResult? {
+        switch matchingAttribute {
+        case let matchingAttribute as PinnedTabsMatchingAttribute:
```

Comment:
Similar to app attributes matcher, handle pinned tabs and custom home page here and use common matcher for the rest.